Add Link component

banciur opened the pull request.

banciur committed:
Add Link component

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+import { connect } from 'react-redux';
+import { routeTo } from 'redux-router-kit';
+
+const LinkView = ({ targetTo, text, path }) =>
```
In the other components I used the naming scheme `SomethingComponent`. I like `SomethingView` better! :D Let's move to this naming scheme. As a separate PR. (NO ACTION REQUIRED)

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+export default connect(
+  null,
+  dispatch => ({
+    targetTo: id => dispatch(routeTo(id)),
```
`id` was called `path` above. It's better to be consistent. `path` sound more descriptive.

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+  (<a
+    tabIndex="0"
+    role="link"
+    onClick={(event) => {
```
We might want to add more properties later (like or `class`, `title`, `target` or even the fancy new `download`). To support this, we can add `...props` to `LinkView` function arguments and add `{...props}` to the element.

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+
+const LinkView = ({ targetTo, text, path }) =>
+  (<a
+    tabIndex="0"
```
We can make the `tabIndex` configurable with default value `0` by adding a property `tabIndex = 0` to `LinkView` arguments. Like so:

```
const LinkView = ({ targetTo, text, path, tabIndex = 0 }) =>
```

recmo replied:
Actually, using `LinkView.defaultProps` might be better. Let me look into this.

recmo replied:
Yes. React doesn't support this new syntax. You need to use `.defaultProps`.

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+const LinkView = ({ targetTo, text, path }) =>
+  (<a
+    tabIndex="0"
+    role="link"
```
I could not find the `role` attribute in the HTML5 specification, what does it do?

banciur replied:
It's a WAI-ARIA attribute. I added role (and tabIndex) as [eslint no-static-element-interactions rule](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md) was triggered.

recmo replied:
Awesome! Eslint is teaching us new things :D

recmo commented on src/components/Link.js:
```diff
@@ -0,0 +1,27 @@
+      event.preventDefault();
+      targetTo(path);
+    }}
+  >{text}</a>);
```
Using the [`children`](https://facebook.github.io/react/docs/composition-vs-inheritance.html) property instead of `text` will make the component even more useful.

recmo reviewed (changes requested):
Looks great, no problems. There are a few changes would make the component even more useful.

banciur committed:
Fixes to Link component based on pull request comments

Make tabIndex attribute settable (and default value) through props
Change text string attribute into children node type
Rename id variable to path
Add passing ...props from function argument to component

banciur committed:
Merge branch 'master' into component_link
